### PR TITLE
feat: add systemd units for docker, containerd, sshd, logind and systemd stubs configs for network and logind

### PR DESCRIPTION
**Pull request by 89luca89**
Useful in case of systemd containers, we want actual services to work
We also add some of the manual config into split packages in order to create these containers more granular and reproducible

### CODE REVIEW

**Comment by 89luca89** [comment]
Contextually bumping containerd to 1.7.23 to please the CI

**Review by dustinkirkland — DISMISSED** [review]

**Review by xnox — CHANGES_REQUESTED** [review]
why hand crafted services used instead of upstream ones?
why none of the hardening options applied in the service units?

**Comment by 89luca89** [comment]
> why hand crafted services used instead of upstream ones? why none of the hardening options applied in the service units?

mostly for user/groups, It's not clear to me how I should "adduser" from apks, given that apko will not run pre/post scriptlets

**Comment by dustinkirkland** [comment]
I'm not clear on this either (adduser in packages).  Would love some help understanding this too!

**Comment by 89luca89** [comment]
I suppose we should keep track of it "manually" and add them in the apko manifests when/if needed
But this makes the image creation a bit more involved than what it should

Maybe a split package (dockerd-user, for example) can be useful, but still I don't know how that should work then

**Comment by 89luca89** [comment]
> > > why hand crafted services used instead of upstream ones? why none of the hardening options applied in the service units?
> > 
> > 
> > mostly for user/groups, It's not clear to me how I should "adduser" from apks, given that apko will not run pre/post scriptlets
> 
> apko has configuration to define users at image build time; you will define stock os users there. Please do not pollute those in packaging. That should help for many of these things. More-over there is a set of stock users we always provide already anyway, and we can expand that static list too - note expanding default static list means we should never reuse the same user or group ids from the reserved range again.

That's ok will make them at apko level
just wanted to be sure I'm not missing something :+1:

**Comment by 89luca89** [comment]
Corrected to use upstream services, with just a patch to fix PATHs where needed

**Comment by 89luca89** [comment]
@dustinkirkland @xnox updated with suggestions (and bumped epochs) can you please take a look? Thanks!

**Comment by 89luca89** [comment]
@xnox @dustinkirkland friendly ping :heart:

File: systemd.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package:
   name: systemd
   version: "256.7"
-  epoch: 4
+  epoch: 5
```
Comment (jamonation) [inline systemd.yaml:4]:
```suggestion
  epoch: 0
```

File: systemd.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package:
   name: systemd
   version: "256.7"
```
Comment (jamonation) [inline systemd.yaml:3]:
```suggestion
  version: "256.8"
```

They just released an hour ago, expected-commit is a2240d1cf9f3515728186c2c98a6f4b64a40e4da

**Comment by 89luca89** [comment]
Thanks @jamonation :heart: 

The CI is blocking containerd as there is an update 1.73.0->2.0.0 seems a bump big enough to deserve it's own PR

**Review by jamonation — APPROVED** [review]
@89luca89 ah yes I see that now too. Well +1 for these changes in any case

**Review by xnox — APPROVED** [review]
Some of these can be improved and get cleaner. The custom setting of path looks like bugs elsewhere in systemd config.

**Comment by xnox** [comment]
Also setting approved to run label for elastic-build result. Or please make your wolfi-dev membership public.

**Pull request merged**